nix-docker

```diff
@@ -0,0 +1,217 @@
+Setup complete.
+EOM
+}
+
+export_env () {
+  log "Setting \$NIX_REMOTE_SYSTEMS to ${remote_sys_conf}"
+  log "Old value: ${NIX_REMOTE_SYSTEMS}"
+  export NIX_REMOTE_SYSTEMS="${remote_sys_conf}"
+  echo
+
+  nix_current_load="/tmp/nix/current-load"
+  log "Setting \$NIX_CURRENT_LOAD to ${nix_current_load}"
+  log "Old value: ${NIX_CURRENT_LOAD}"
+  mkdir -p "${nix_current_load}"
+  chmod a+rwX "${nix_current_load}"
+  export NIX_CURRENT_LOAD="/tmp/nix/current-load"
+  echo
+
+  build_remote="${HOME}/.nix-profile/libexec/nix/build-remote.pl"
+  log "Setting \$NIX_BUILD_HOOK to ${build_remote}"
+  log "Old value: ${NIX_BUILD_HOOK}"
+  export NIX_BUILD_HOOK="${build_remote}"
+  echo
+}
+
+do_init () {
+  ssh_config="${HOME}/.ssh/config"
+
+  # Default values
+  working_dir="${HOME}/${working_dir_name}"
+  docker_name="nix-docker"
+  key_file="./ssh/insecure_rsa"
+  port_num="3022"
+
+  while [[ $# -gt 0 ]]; do
+    case "${1}" in
+      -d|--dir)
+        working_dir=${2}
+        shift 2
+        ;;
+      -n|--name)
+        docker_name=${2}
+        shift 2
+        ;;
+      -k|--key)
+        key_file=${2}
+        shift 2
+        ;;
+      -p|--port)
+        port_num=${2}
+        shift 2
+        ;;
+    esac
+  done
+
+  # Set up environment
+  mkdir -p "${working_dir}"
+
+  # Copy SSH keys to working directory
+  cp "${key_file}" "${working_dir}"
+  cp "${key_file}.pub" "${working_dir}"
+
+  key_file_name=$(basename "${key_file}")
+  ssh_id_file="${working_dir}/${key_file_name}"
+  chmod 600 "${ssh_id_file}"
+
+  # Check to see if ~/.ssh/config exists - if not, create it
+  [[ -f "${ssh_config}" ]] || touch "${ssh_config}"
+
+  # Check if SSH config file has an entry for our Docker machine - if not, add it
+  if ! grep "${docker_name}\$" "${ssh_config}" > /dev/null; then
+    log "Adding an entry to ${ssh_config} for ${docker_name}."
+    cat >> "${ssh_config}" <<CONF
+
+Host ${docker_name}
+  User root
+  HostName 127.0.0.1
+  Port ${port_num}
+  IdentityFile ${ssh_id_file}
+CONF
+  else
+    log "SSH configuration already contains an entry for ${docker_name}."
+  fi
+  echo
+
+  # Start docker container
+  log "Starting docker container: ${docker_name}"
+  log "Note: This may fail if the container was already created."
+  docker run --restart always --name "${docker_name}" -d -p ${port_num}:22 lnl7/nix:ssh
+  echo
+
+  # Write remote systems configuration
+  remote_sys_conf="${working_dir}/remote-systems.conf"
+  log "Writing remote systems configuration to ${remote_sys_conf}"
+
+  cat > "${remote_sys_conf}" <<CONF
+${docker_name} x86_64-linux ${ssh_id_file} 1
+CONF
+  echo
+
+  # Test connection
+  log "Running SSH test"
+  ssh "${docker_name}" echo "SSH connection succeeded!" || log "SSH connection failed."
+  echo
+
+  # Copy SSH key to container
+  log "Copying SSH key to container"
+  scp "${key_file}.pub" nix-docker:/root/.ssh/authorized_keys
+
+  # Export environment
+  export_env
+
+  remove_instructions
+  finish_message
```

Comment:
Also in some cases that I can't seem to isolate the script does not give back the prompt at the end, which is one of the reasons why I added this line.. not sure if you have any ideas.